src/components/Post.tsx

```diff
@@ -0,0 +1,244 @@
+import {
+  Image,
+  ImageBackground,
+  LayoutChangeEvent,
+  Pressable,
+  ScrollView,
+  StyleSheet,
+  Text,
+  View,
+  useWindowDimensions,
+} from 'react-native';
+import React, {useEffect, useState} from 'react';
+import LinearGradient from 'react-native-linear-gradient';
+import IconButton from './IconButton';
+import {useNavigation} from '@react-navigation/native';
+import createImageInfo from '../assets/home-mock-data';
+import type {PostType} from '../assets/posts-mock-data';
+import Animated, {
+  Easing,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+
+interface PostProps {
+  item: PostType;
+}
+
+export default function Post({item}: PostProps) {
+  const {height} = useWindowDimensions();
+  const maxHeroHeight = height - (64 + 60 + 40 + 60 + 10); // right under the header
```

Comment:
change these values into CONSTANTS to become more readable & easier to maintain, please